crates/autopilot/src/solvable_orders.rs

```diff
@@ -795,6 +824,21 @@ impl OrderFilterCounter {
         filtered_orders.into_keys().collect()
     }
 
+    /// Creates a new checkpoint based on the found invalid orders.
+    fn checkpoint_by_invalid_orders(&mut self, reason: Reason, invalid_orders: &[OrderUid]) {
+        *self.counts.entry(reason).or_default() += invalid_orders.len();
+        for order_uid in invalid_orders {
+            self.orders.remove(order_uid).unwrap();
+        }
+        if !invalid_orders.is_empty() {
```

Comment:
Seems like it would be a bit nicer to have this early return at the start of the function.